llpc/context/llpcCompiler.cpp

```diff
@@ -832,11 +832,27 @@ Result Compiler::buildPipelineWithRelocatableElf(Context *context, ArrayRef<cons
 static bool hasUnrelocatableDescriptorNode(const ResourceMappingNode *nodes, unsigned nodeCount) {
   for (unsigned i = 0; i < nodeCount; ++i) {
     const ResourceMappingNode *node = nodes + i;
-    if (node->type == ResourceMappingNodeType::DescriptorTableVaPtr) {
-      if (hasUnrelocatableDescriptorNode(node->tablePtr.pNext, node->tablePtr.nodeCount))
-        return true;
-    } else if (node->type == ResourceMappingNodeType::DescriptorBufferCompact) {
+    switch (node->type) {
+    case ResourceMappingNodeType::DescriptorTableVaPtr: {
+      const ResourceMappingNode *startInnerNode = node->tablePtr.pNext;
+      const ResourceMappingNode *endInnerNode = startInnerNode + node->tablePtr.nodeCount;
+      for (const ResourceMappingNode *innerNode = startInnerNode; innerNode != endInnerNode; ++innerNode) {
+        if (innerNode->type == ResourceMappingNodeType::DescriptorBufferCompact)
+          // The code to handle a compact descriptor cannot be easily patched, so relocatable shaders assume there are
+          // no compact descriptors.
+          return true;
+      }
+    } break;
```

Comment:
Could you move the break to {} body. That is

case XXX: {
   // Code piece
  break;
}
case YYY:

Reply:
done.